docs/how-to-guides/upgrade-rook-ceph.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Upgrading Rook Ceph
+weight: 10
+---
+
+## Introduction
+
+[Rook Ceph](../configuration-reference/components/rook.md) is one of the storage providers of
+Lokomotive. With a distributed system as complex as Ceph, the upgrade process is not trivial. This
+document enlists steps on how to perform the upgrade and how to monitor this process.
+
+## Prerequisites
+
+- A Lokomotive cluster accessible via `kubectl`.
+
+- Upgrade process should not be disruptive, but it is recommended to schedule a downtime for the
+  applications consuming the Rook Ceph PVs and make sure to take a backup of the data before
+  starting the below upgrade procedure in case some problem arises. Read more about the [backup
+  process](backup-rook-ceph-volumes.md).
+
+## Steps
+
+Following steps are inspired by [`rook`](https://rook.io/docs/rook/master/ceph-upgrade.html) docs.
+
+### Step 1: Ensure `AUTOSCALE` is set to `on`
+
+Start a shell in the toolbox pod as specified in [this
+doc](rook-ceph-storage.md#enable-and-access-toolbox) and run the following command:
+
+```console
+# ceph osd pool autoscale-status | grep replicapool
+POOL           SIZE  TARGET SIZE  RATE  RAW CAPACITY   RATIO  TARGET RATIO  EFFECTIVE RATIO  BIAS  PG_NUM  NEW PG_NUM  AUTOSCALE
+replicapool      0                 3.0         3241G  0.0000                                  1.0      32              on
+```
+
+Ensure that the `AUTOSCALE` column outputs `on` and not `warn`. If the output of the `AUTOSCALE`
+column says `warn`, then run the command below, to make sure that pool autoscaling is enabled. It is
+required to ensure that the placement groups scale up as the data in the cluster increases.
+
+```bash
+ceph osd pool set replicapool pg_autoscale_mode on
+```
+
+### Step 2: Watch
+
+Watch events, updates and pods.
+
+#### Step 2.1: Ceph status
+
+Leave the following running in the toolbox pod:
+
+```bash
+watch ceph status
+```
+
+Ensure that the output says that `health:` is `HEALTH_OK`. Match the output such that everything
+looks as explained in the [rook upgrade
+docs](https://rook.io/docs/rook/master/ceph-upgrade.html#status-output).
+
+> **IMPORTANT**: Don't proceed further if the output is anything other than `HEALTH_OK`.
+
+During the ongoing upgrade and after completion, make sure that the output stays in `HEALTH_OK`
+state. If the cluster is more than 60% full, then the output can sometimes turn into `HEALTH_WARN`.
```

Comment:
So what happens if during the upgrade it shows `HEALTH_WARN` is it ok if it doesn't get full or does everything explode after that? What advice can we give to the user?

Reply:
HEALTH_WARN is ok to be in. TBH the user does not have control over the upgrade process and if the upgrade fails for whatever reason we can have look into it case by case basis. This upgrade doc should reflect such cases if there are any coming from upstream.

Reply:
Then I guess we shouldn't say "make sure that the output stays in `HEALTH_OK`" because it's not actionable for the user. I'll make a suggestion.